pkgs/development/python-modules/gpt4all/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, requests
+, tqdm
+, cmake
+, pytestCheckHook
+}:
+
+let
+  version = "1.0.8";
+  src = fetchFromGitHub {
+      owner = "nomic-ai";
+      repo = "gpt4all";
+      rev = "39acbc837816ea9c7673bfe5cf83aeaed95b2c5f";
+      hash = "sha256-aamPt4IrzA+07N0gxt1F+Cc9/qiNL8g4CGyYjaWx144=";
+      fetchSubmodules = true;
+    };
+
+  deps = stdenv.mkDerivation rec {
+    inherit src;
+    name = "gpt4all-backend";
+    sourceRoot = "${src.name}/gpt4all-backend";
+
+    nativeBuildInputs = [
+      cmake
+    ];
+
+    installPhase = ''
+      mkdir -p $out
+      install -m 0644 *.so $out
+    '';
+  };
+in
+buildPythonPackage rec {
+  inherit src version;
+
+  pname = "gpt4all";
+  sourceRoot = "${src.name}/gpt4all-bindings/python";
+  disabled = pythonOlder "3.8";
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  propagatedBuildInputs = [
+    requests
+    tqdm
+  ];
+
+  patches = [ ./disable-copy.patch ];
```

Comment:
```suggestion
  # It'd be nice if upstream was aware of this discomfort..  
  patches = [ ./disable-copy.patch ];
```

Reply:
> `# It'd be nice if upstream was aware of this discomfort..`

Hmm I'm surprised this was my code suggestion back then :) It'd be nice if you had opened an issue upstream and linked to it.